scanner/mappers/mappers.go

```diff
@@ -306,6 +334,55 @@ func fixedInVersion(v *claircore.Vulnerability) string {
 	return fixedIn
 }
 
+// severityValues contains information that can be encoded in the severity of vulnerabilities.
+type severityValues struct {
+	severity string
+	v2Vector string
+	v2Score  float32
+	v3Vector string
+	v3Score  float32
+}
+
+// parseSeverity parses the severity information encoded in the severity field.
+// The information is distribution dependant.  If errors are found in the
+// encoded information, partial values are returned with an error.
+func parseSeverity(v *claircore.Vulnerability) (s severityValues, err error) {
+	var did string
+	if v.Dist != nil {
+		did = v.Dist.DID
+	}
+	switch did {
+	case "rhel":
+		var q url.Values
+		if q, err = url.ParseQuery(v.Severity); err != nil {
+			err = fmt.Errorf("invalid RHEL severity: %w", err)
+			break
+		}
+		s.severity = q.Get("severity")
+		// Look for CVSS fields, when at least one vector is found, we consider the CVSS
+		// information exists, but return partial results in case of failures.
+		errList := errorhelpers.NewErrorList("invalid RHEL CVSS")
```

Comment:
nit: I'm not a huge fan of this library and would prefer just using `errors.Join` (thought that's, admittedly more work)

Reply:
I like it.

Reply:
I am resolving this.